main.go

```diff
@@ -41,10 +41,16 @@ func startLogServe(stream *stream.Stream) {
 		// URL and comparing the reminder to the accessToken, ensuring a URL pattern
 		// /log/<accessToken>
 		if r.URL.String()[5:] != accessToken {
-			w.Header().Set("Content-Type", "text/plain; charset=utf-8")
-			w.Header().Set("Access-Control-Allow-Origin", "*")
+			writeHeaders(w, r)
 			w.WriteHeader(401)
 			fmt.Fprint(w, "Access denied")
+		} else if r.Method == "HEAD" {
```

Comment:
I think this should be `OPTIONS` rather than `HEAD`. See [spec](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests).